src/main/java/net/snowflake/client/jdbc/cloud/storage/SnowflakeGCSClient.java

```diff
@@ -693,6 +698,31 @@ public void upload(
         "Unexpected: upload unsuccessful without exception!");
   }
 
+  /**
+   * Upload file with down scoped token.
+   *
+   * @param remoteStorageLocation storage container name
+   * @param destFileName file name on remote storage after upload
+   * @param contentEncoding Object's content encoding. We do special things for "gzip"
+   * @param metadata Custom metadata to be uploaded with the object
+   * @param content File content
+   */
+  private void uploadWithDownScopedToken(
+      String remoteStorageLocation,
+      String destFileName,
+      String contentEncoding,
+      Map<String, String> metadata,
+      InputStream content) {
+    BlobId blobId = BlobId.of(remoteStorageLocation, destFileName);
```

Comment:
Please check JDBC code. There is third parameter to add otherwise it was causing some issue.

Reply:
The line 716 is for line `646`. It's only 2 parameters. I don't see 3rd parameter is used. Can you refer to the code?

Reply:
I was talking about this API call: blob.downloadTo(
              localFile.toPath(), Blob.BlobSourceOption.shouldReturnRawInputStream(true));. Make sure you have this third parameter in the call, otherwise it was failing in case of JDBC test.